### PR TITLE
Socket Message

### PR DESCRIPTION
**Pull request by a-givertzman**

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -12,9 +12,10 @@ dependencies:
     sdk: flutter
   logging: ^1.3.0
   hmi_core:
-      git:
-        url: https://github.com/a-givertzman/hmi_core.git
-        ref: 2.0.0
+    path: ../hmi_core/
+    # git:
+    #   url: https://github.com/a-givertzman/hmi_core.git
+    #   ref: 2.1.0
```
Comment (Minyewoo) [inline pubspec.yaml:18]: Don't forget to change it back

File: pubspec.yaml
Comment (Minyewoo) [inline pubspec.yaml:1]: Don't forget to increase the package version

File: lib/src/api_client/message/parse_size.dart [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import 'package:ext_rw/src/api_client/message/field_kind.dart';
+import 'package:ext_rw/src/api_client/message/field_size.dart';
+import 'package:ext_rw/src/api_client/message/message_parse.dart';
+import 'package:hmi_core/hmi_core_option.dart';
+import 'package:hmi_core/hmi_core_result.dart';
+///
+///
+class ParseSize implements MessageParse<List<int>, Option<(FieldKind, FieldSize, List<int>)>> {
+  final MessageParse<dynamic, Option<(FieldKind, List<int>)>> _field;
```
Comment (nyaneet) [inline lib/src/api_client/message/parse_size.dart:9]: See comment above

File: lib/src/api_client/message/message_build.dart [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  ///   - [data] - Data of length specified in [size] will be fetched from the socket
+  MessageBuild({
+    required this.syn,
+    required this.kind,
+    required this.size,
+    required this.data,
+  });
+  ///
+  /// Returns message built according to specified fields and [bytes]
+  List<int> build(List<int> bytes) {
```
[the remark this answers is not in the document]
Comment (nyaneet) [reply lib/src/api_client/message/message_build.dart:61]: Agree with @Minyewoo.
Prefer to use `UInt8List` for package data as more space- and time-efficient implementation of `List<int>`.

[the remark this answers is not in the document]
Comment (a-givertzman) [reply lib/src/api_client/message/message_build.dart:61]: At the moment not sure it's totally better, 
- May be in the future we'll have some tests, wich can highlight a difference.
- Also this [post can be useful](https://stackoverflow.com/a/22070044) can be useful

File: lib/src/api_client/message/message_parse.dart [hunk 4]
```diff
@@ -0,0 +1,51 @@
+///
+/// # Messages received over socket.
+/// 
+/// - Data can be encoded using varius data `Kind`, `Size` and payload Data
+/// 
+/// - Message format
+///     Field name | Start | Kind |  Size  | Data |
+///     ---       |  ---  | ---  |  ---   | ---  |
+///     Data type |  u8   | u8   | u32    | [u8; Size] |
+///     Value     |  22   | StringValue | xxx    | [..., ...]  |
```
[the remark this answers is not in the document]
Comment (a-givertzman) [reply lib/src/api_client/message/message_parse.dart:10]: Afraid it is not exactly the same as just `toBytes` and `fromBytes`.

File: pubspec.yaml (hunk 1 above)
Comment (a-givertzman) [reply pubspec.yaml]: 2.1.2 for now

**Pull request merged**